tests/cargo-kani/build-rs-plus-host-with-kani-proofs/constants/src/lib.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// From https://github.com/model-checking/kani/issues/3101
+
+pub const SOME_CONSTANT: u32 = 42;
```

Comment:
I would suggest to enhance this test a bit. You can define a different value for this constant based on whether this is `kani_host`, `kani` or none. something like:

```suggestion
#[cfg(not(any(kani, kani_host)))]
pub const SOME_CONSTANT: u32 = 0;
#[cfg(kani)]
pub const SOME_CONSTANT: u32 = 1;
#[cfg(kani_host)]
pub const SOME_CONSTANT: u32 = 2;
```

So in the build script, you can add:
```rust

#[cfg(not(kani_host))]
assert_eq!(constants::SOME_CONSTANT, 0);
#[cfg(kani_host)]
assert_eq!(constants::SOME_CONSTANT, 2);
```

and inside the harness, you can add:
```rust
assert_eq!(constants::SOME_CONSTANT, 1);
```